tools/quickstart/quickstart/utils.py

```diff
@@ -0,0 +1,143 @@
+        self.password_file = password_file
+        self.address_file = address_file
+        self.keystore_file = keystore_file
+
+    def store(self, account, password):
+        # Instead of just copying the file, we encrypt it again since the
+        # file itself may use scrypt as key derivative function which
+        # parity can't handle
+        json_account = account.encrypt(password, kdf="pbkdf2")
+
+        os.makedirs(os.path.dirname(os.path.abspath(self.keystore_file)), exist_ok=True)
+        with open(self.keystore_file, "w") as f:
+            f.write(json.dumps(json_account))
+
+        with open(self.password_file, "w") as f:
+            f.write(password)
+
+        with open(self.address_file, "w") as f:
+            f.write(account.address)
+
+
+def is_wrong_password_error(err):
+    return type(err) is ValueError and str(err) == "MAC mismatch"
+
+
+def get_keystore_path() -> str:
+    click.echo(
+        "Please enter the path to the keystore file to import.\nIf you started the process via the quickstart scipt, "
+        "please consider that you are running within a Docker virtual file system. "
+        "You can access the current working directory via '/data'. "
+        "(e.g. './my-dir/keystore.json' becomes '/data/my-dir/keystore.json')"
+    )
+
+    while True:
+        raw_path = click.prompt("Path to keystore", default="./account-key.json")
+        absolute_path = Path(raw_path).expanduser().absolute()
+
+        if os.path.isfile(absolute_path) and os.access(absolute_path, os.R_OK):
+            return absolute_path.as_posix()
+
+        else:
+            click.echo(
+                "The given path does not exist or is not readable. "
+                "Please try entering it again."
+            )
+
+
+def read_private_key() -> str:
+    while True:
+        private_key = click.prompt("Private key (hex encodded)", hide_input=True)
+
+        if re.fullmatch("[0-9a-fA-F]{64}", private_key):
+            return private_key
+
+        click.echo(
+            "The private key must be entered as a hex encoded string. Please try again."
+        )
+
+
+def read_encryption_password() -> str:
+    click.echo(f"Please enter a password to encrypt the private key.")
+
+    while True:
+        password = click.prompt("Password", hide_input=True)
+        password_repeat = click.prompt("Password (repeat)", hide_input=True)
+
+        if password == password_repeat:
+            return password
+
+        click.echo("Passwords do not match. Please try again.")
+
+
+def read_decryption_password(keyfile_dict) -> Tuple[Account, str]:
+    click.echo("Please enter the password to decrypt the keystore.")
+
+    while True:
+        password = click.prompt("Password", hide_input=True)
+
+        try:
+            account = Account.from_key(Account.decrypt(keyfile_dict, password))
+            return account, password
+
+        except Exception as err:
+            if is_wrong_password_error(err):
```

Comment:
better use `except ValueError as err` and check `is_wrong_password` only there

Reply:
Again this is "old" code.
But what is your point? This would make the code just more complicated. Now I have two cases. Then I will have three, where two are the same. I don't want to let the error be raised uncaught to the end-user.

```python
except ValueError as err:
    if str(err) == "MAC mismatch":
        # case a
    else:
        # case b
except Exception:
     # also case b
```

Reply:
You could add another level of nesting to get rid of the dublication:

```Py
try:
    try:
        do_stuff()
    except ValueError as value_error:
        if is_right_error(value_error):
             handle_it()
        else:
             raise
except Exception as exception:
    print_error()
```

My point is that catching plain exceptions is generally a bad idea and one should avoid it if possible. I guess the main problem is that the keystore library doesn't raise more specific exceptions, but as least for in the cases in which it does we should try to use them.

But if it's preexisting code I'm fine with keeping it.

Reply:
Catching `Exception` is bad practice yes, because you might hide actual errors by that. What exception do you try to catch by that?

Reply:
But as long as you stop the program and print the error it is actually fine.

Reply:
@jannikluhn I see your point and I will definitely keep it in mind for better future development. :+1: 
I think Ralf has thought about it as well and this is what he has end up with. Since the suggested solution does not look appealing to me, I would rather not change it withing this PR. 


---

> What exception do you try to catch by that?

@berndbohmeier not 100% sure how you question is targeted. For me it is detecting the error of a wrong password for the to import keystore. I left it untouched, because I don't know how to do it better. :man_shrugging: